### PR TITLE
[WNMGDS-1180] mgov icons

### PR DESCRIPTION
**Pull request by scheul93**
## Summary
Updating mgov icons to follow pattern of icons in core

### Added
1. A `RoundedStarIcon` component since mgov star icons are different than core.
2. A `HamburgerIcon` to core that can be used in mgov

### Changed
Moved the following Icons into an Icon directory for Mgov. These components will eventually be removed:
* Caret Icon
* Checkmark Icon
* Close Icon
* Hamburger Icon
* No Icon

### Deprecated
Marked the following components as deprecated by adding a `console.error` in the component as well as an alert on the documentation page
* Caret Icon
* Checkmark Icon
* Close Icon
* Hamburger Icon
* No Icon

### Removed
All files except the `.svg` file for the following directories in mgov
* Caret
* Checkmark
* Close
* Hamburger
* No

## How to test
I intentionally have not updated the HTML examples for the icons I updated because it's kind of an easy way to see if the icons are the same! (I can do that in a separate PR)

Run the medicare DS locally by running `yarn build`, `yarn start:medicare`

Look at icon pages where I updated the icon to use the core icon version. Note the deprecation warnings:
* [Caret](http://localhost:3000/components/caret-icon/)
* [Checkmark](http://localhost:3000/components/checkmark-icon/)
* [Close](http://localhost:3000/components/close-icon/)
* [Hamburger](http://localhost:3000/components/hamburger-icon/)
* [No](http://localhost:3000/components/No/)

Compare the `Stars` component to the live site and ensure they are the same: [demo branch](http://localhost:3000/components/stars-icons/) vs [live site](https://cmsgov.github.io/design-system/medicare/components/stars-icons/)

Look at the [Icons page](http://localhost:3000/components/icon/) and note the hamburger menu has been added to the list of core icons & the mgov-specific icons have their own section

Note: This PR does not include moving svgs that are "logos".

### CODE REVIEW

**Comment by forrestbaer** [comment]
I was unable to build (no typedefs for MenuIconThin and HhsLogo from core) without bumping the core version to 2.11.0-beta.1 in the medicare package.json

image

**Comment by forrestbaer** [comment]
All the other changes are testing fine for me otherwise!

**Review by zarahzachz — COMMENTED** [review]
Would it make sense to also include the deprecation notices in the notes section of [SvgIcon list of icons](https://github.com/CMSgov/design-system/blob/00fd6b2c6dcce8a33364af8678c1feb358ba382a/packages/design-system-docs/src/pages/components/Icon/IconComponent.example.jsx)?

**Comment by zarahzachz** [comment]
Question about using `console.error` - do we know if teams' builds are gonna get borked by including a `console.error` on our end? Devs have scolded me about this in the past - I think the issue was dumping a bunch of errors in their console and they didn't have warning or bandwidth to clean them up for that sprint. But I don't know if this is an issue for us.

**Comment by scheul93** [comment]
> SvgIcon list of icons

I'm not sure I understand. All the icons in that list are new and none are being deprecated. The deprecated icons are mgov-specific and were components already in the mgov design system. They each have individual doc pages on the mgov doc site

**Comment by scheul93** [comment]
> Question about using `console.error` - do we know if teams' builds are gonna get borked by including a `console.error` on our end? Devs have scolded me about this in the past - I think the issue was dumping a bunch of errors in their console and they didn't have warning or bandwidth to clean them up for that sprint. But I don't know if this is an issue for us.

I'm interested in @pwolfert 's thoughts. I know in the past, there's been hesitation in removing deprecated props/features because we deprecation notices were just warnings and maybe they weren't as noticeable. I figured this was a way to really let teams know that they need to stop using these components and potentially speed up our deprecation timeline

**Comment by forrestbaer** [comment]
The stars look fine, but I'm seeing a 'heavier' checkmark and caret in mine between the html & react renders

image
image

Modal icon looks right now.

File: packages/design-system/src/components/Icons/StarIcon.tsx [hunk 1]
```diff
@@ -15,7 +14,9 @@ const defaultProps = {
 function StarIcon(props: StarIconProps): React.ReactElement {
   const iconCssClasses = classNames(
     'ds-c-icon--star',
-    { 'ds-c-icon--star-filled': props.isFilled },
+    {
+      'ds-c-icon--star-filled': props.isFilled,
+    },
```
Comment (pwolfert) [inline packages/design-system/src/components/Icons/StarIcon.tsx:19]: This file doesn't seem to have really changed except for whitespace

File: packages/ds-medicare-gov/docs/src/pages/components/HamburgerSymbol/HamburgerSymbol.docs.scss [hunk 2]
```diff
@@ -1,6 +1,16 @@
 /*
 Hamburger Icon
 
+<div class="ds-c-alert ds-c-alert--error">
+    <div class="ds-c-alert__body">
+      <h2 class="ds-c-alert__heading">Deprecation Notice</h2>
+      <p class="ds-c-alert__text">
+        This component has been marked as 'deprecated' and will be removed in a future release. Please use the component `HamburgerIcon` with CSS class 'ds-c-icon-color--primary' instead.
+        See details for `HamburgerIcon` on the <a href="https://cmsgov.github.io/design-system/medicare/components/icon/">Icons page</a>
```
Comment (pwolfert) [inline packages/ds-medicare-gov/docs/src/pages/components/HamburgerSymbol/HamburgerSymbol.docs.scss:9]: Does this need to say `HamburgerIconThin`?

File: packages/ds-medicare-gov/src/components/Icons/Caret.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  className?: string;
+  up?: boolean;
+  left?: boolean;
+  right?: boolean;
+}
+
+const Caret: FunctionComponent<CaretProps> = ({ className, up, left, right }) => {
+  console.error(
+    `[Deprecated]: Please use the <ArrowIcon /> component instead. This component will be removed in a future release.`
+  );
```
Comment (pwolfert) [inline packages/ds-medicare-gov/src/components/Icons/Caret.tsx:15]: 👍

**Review by pwolfert — APPROVED** [review]
Overall, this LGTM 👍

**Comment by pwolfert** [comment]
>> Question about using console.error - do we know if teams' builds are gonna get borked by including a console.error on our end? Devs have scolded me about this in the past - I think the issue was dumping a bunch of errors in their console and they didn't have warning or bandwidth to clean them up for that sprint. But I don't know if this is an issue for us.

> I'm interested in @pwolfert 's thoughts. I know in the past, there's been hesitation in removing deprecated props/features because we deprecation notices were just warnings and maybe they weren't as noticeable. I figured this was a way to really let teams know that they need to stop using these components and potentially speed up our deprecation timeline

@zarahzachz and @scheul93, we probably need to continue this general conversation in the context of some documented deprecation process (which is in the works, right? do we need to create a ticket to track it?), but in this particular case I would argue these are safe because we have one team using mgov styles, and we aren't actually planning a real release (just a series of beta releases) until the end of the PI, at which point we will be looking at fully deprecating them. Or maybe we won't fully deprecate them, but we'll be releasing breaking changes around that time anyway.

**Comment by zarahzachz** [comment]
> > SvgIcon list of icons
> 
> I'm not sure I understand. All the icons in that list are new and none are being deprecated. The deprecated icons are mgov-specific and were components already in the mgov design system. They each have individual doc pages on the mgov doc site

Oh I must've misunderstood - I thought you were adding deprecation notices for some of the icons found on that main SvgIcon list. Disregard my comment then!

**Pull request merged**